nevergrad/functions/stsp/core.py

```diff
@@ -30,6 +31,8 @@ def _simulate_stsp(self, x: np.ndarray) -> float:
         output = np.sqrt((x[0] - x[-1]) ** 2 + (y[0] - y[-1]) ** 2) + sum(
             np.sqrt((x[i] - x[i + 1]) ** 2 + (y[i] - y[i + 1]) ** 2) for i in range(self.dimension - 1)
         )
+        if self.complex:
+            return float(output) + sum((order[i] - order[i+1]) ** 2 for i in range(self.dimension - 1)) / np.sqrt(self.dimension + 1.)
```

Comment:
please use numpy functions and not for loops, this is too inefficient
you just need `np.diff`